packages/grafana-ui/src/components/Logs/LogRowContextProvider.tsx

```diff
@@ -62,16 +63,33 @@ export const getRowContexts = async (
       const data: any[] = [];
       for (let index = 0; index < dataResult.data.length; index++) {
         const dataFrame = toDataFrame(dataResult.data[index]);
-        const timestampField: Field<string> = dataFrame.fields.filter(field => field.name === 'ts')[0];
+        const fieldCache = new FieldCache(dataFrame);
+        const timestampField: Field<string> = fieldCache.getFieldByName('ts')!;
+        const idField: Field<string> | undefined = fieldCache.getFieldByName('id');
 
         for (let fieldIndex = 0; fieldIndex < timestampField.values.length; fieldIndex++) {
-          const timestamp = timestampField.values.get(fieldIndex);
-
-          // We need to filter out the row we're basing our search from because of how start/end params work in Loki API
-          // see https://github.com/grafana/loki/issues/597#issuecomment-506408980
-          // the alternative to create our own add 1 nanosecond method to the a timestamp string would be quite complex
-          if (timestamp === row.timestamp) {
-            continue;
+          // TODO: this filtering is datasource dependant so it will make sense to move it there so the API is
+          //  to return correct list of lines handling inclusive ranges or how to filter the correct line on the
+          //  datasource.
+
```

Comment:
We should probably create an issue for this, so we won't forget about moving it to datasource.